 design/README.MD

```diff
@@ -0,0 +1,456 @@
+and that the correct duals are connected together.
+
+It is much easier if we can write our tests in Dafny.
+Then we don't need to write new tests for each runtime.
+This is where [wrapping a library](#wrapped-library) comes in.
+
+We want the test in Dafny and we want to test the conversions.
+by building a project we can control the Dafny implementation.
+By wrapping this project we can call it from Dafny
+and pass through the type conversions.
+This sandwiches the conversions between two functions we control.
+Namely, the implementation and the test.
+Many of these are already written in the TestModels directory.
+
+For the simple types: string, number, et al
+the implementation can not just echo back values.
+This is because it is possible that the conversion function
+do not produce rational input in the middle.
+This is an edge case, but it is easy enough to verify.
+
+Each simple type has smithy operations that:
+* returns back what is given.
+* only succeed if passed a specific value.
+* will only return a constant value
+
+We have one last subtlety to consider.
+Since the tests and implementation are both in Dafny
+we can clearly see the Dafny representation is reasonable.
+But this does not tell us that the native representation is reasonable.
+
+Consider a string.
+This string is represented under the hood as bytes
+and these bytes have an encoding.
+In `smithy-dafny` we represent this string as UTF8 bytes.
+If the conversion was to simply change the encoding to USC2
+then all our tests would pass.
+But looking at the native string it would be drastically different.
+
+This is a degenerate example but should illustrate the point.
+At this time we do not have any such degenerate tests written.
+This is because maintaining such coverage across multiple runtime
+is exceedingly difficult.
+It might be that a specification of such tests in conjunction with duvet
+might be a possible solution.
+
+## Project Management
+
+Neither Dafny nor `smithy-dafny` provide project management features.
+They are not terribly opinionated
+about how native project should be set up.
+
+`smithy-dafny` can take a smithy model
+and output the generated native code anywhere.
+A `runtimes` directory exists in a `smithy-dafny` project
+and inside that there is a directory for each native project.
+
+Looking at the TestModels directory you can see examples.
+The native project can be set up however is best for the native language.
+However the native project needs to bind the following things together
+
+1. `smithy-dafny` generated Polymorph code
+1. Translated Dafny implementation code
+1. Translated Dafny test code
+1. Native `smithy-dafny` dependencies
+    * Dafny runtime
+    * Dafny conversion libraries
+    * Dafny project dependencies
+    * Native dependencies - 
+
+In the TestModels directory there is a SharedMakefile
+that can be used to provide useful targets for Dafny projects.
+
+Here is an example project structure.
+
+```mermaid
```

Comment:
This one shouldn't be a diagram, just ascii art. e.g. the directory structure snippet from https://github.com/smithy-lang/smithy-typescript?tab=readme-ov-file#using-smithy-typescript-with-gradle